src/components/Tree/TreeItem.tsx

```diff
@@ -53,28 +55,7 @@ export const TreeItem = ({ id, sort, label, contentComponent, onSelect, onDrop,
     let enhancedChildren: ReactNode = childrenArray;
 
     if (draggable) {
-        enhancedChildren = React.Children.map(children, (child, index) => {
-            if (!child) {
-                return <></>;
-            }
-
-            return React.cloneElement(
-                <>
-                    {index === 0 && (
-                        <DropZone
-                            data={{
-                                targetItem: { id, sort },
-                                position: DropZonePosition.Before,
-                            }}
-                            onDrop={handleDrop}
-                            treeId={treeId}
-                        />
-                    )}
-
-                    {child}
-                </>,
-            );
-        });
+        enhancedChildren = enhanceChildren(treeId, handleDrop, children);
```

Comment:
enhanceChildren rename to useDraggableEnhancedChildren. Also do not pass treeId and handleDrop as arguments, just reuse useTreeContext hook inside